zoo/board_games/chess/envs/chess_lightzero_env.py

```diff
@@ -0,0 +1,466 @@
+
+        if self.current_player == 1:
+            reward = -reward
+
+        info = {}
+        if done:
+            info['eval_episode_return'] = reward
+
+        action_mask = np.zeros(4672, dtype=np.int8)
+        action_mask[chess_utils.legal_moves(self.board)] = 1
+
+        obs = {
+            'observation': self.observe(self.current_player_index)['observation'],
+            'action_mask': action_mask,
+            'board': self.board.fen(),
+            'current_player_index': 1 - self.current_player_index,
+            'to_play': self.next_player
+        }
+
+        self.current_player = self.next_player
+
+        return BaseEnvTimestep(obs, reward, done, info)
+
+    @property
+    def current_player(self):
+        return self._current_player
+
+    @property
+    def current_player_index(self):
+        return 0 if self._current_player == 1 else 1
+
+    @property
+    def next_player(self):
+        return self.players[0] if self.current_player == self.players[1] else self.players[1]
+
+    @current_player.setter
+    def current_player(self, value):
+        self._current_player = value
+
+    def random_action(self):
+        action_list = chess_utils.legal_moves(self.board)
+        return np.random.choice(action_list)
+
+    def bot_action(self):
+        # Get all legal moves
+        legal_moves = list(self.board.legal_moves)
+        legal_actions = chess_utils.legal_moves(self.board)
+
+        # Create a temporary board to simulate moves
+        temp_board = self.board.copy()
+
+        best_action = None
+        best_score = -np.inf
+
+        # Evaluate each legal move
+        for action, move in zip(legal_actions, legal_moves):
+            # Simulate the move
+            temp_board.push(move)
+
+            # Evaluate the board after the move
+            score = self.evaluate_board(temp_board)
+
+            # Update best_action if the current move has a higher score
+            if score > best_score:
+                best_action = action
+                best_score = score
+
+            # Undo the move to return to the original state
+            temp_board.pop()
+
+        # Return the action with the highest score
+        return best_action
+
+    def evaluate_board(self, board):
+        # Evaluate the board based on rules
+
+        # Check if the game is over
+        outcome = board.outcome()
+        if outcome is not None:
+            # If the game is over, return a definitive score
+            if outcome.winner == chess.WHITE:
+                return 10000  # White wins
+            elif outcome.winner == chess.BLACK:
+                return -10000  # Black wins
+            else:
+                return 0  # Draw
+
+        # If the game is not over, calculate a score based on various factors
+        score = 0
+
+        # 1. Assign points based on the value of pieces
+        # Piece values: Queen 9, Rook 5, Knight 3, Bishop 3, Pawn 1
+        piece_values = {
```

Comment:
I think these codes are the same as ``chess_env.py``. Maybe you can merge them.

Reply:
Merged now, thanks!